pkg/controller/shoot/shoot_control_reconcile.go

```diff
@@ -69,6 +70,8 @@ func (c *defaultControl) reconcileShoot(o *operation.Operation, operationType ga
 		deployETCD                              = f.AddTask(hybridBotanist.DeployETCD, defaultRetry, deploySecrets, waitUntilBackupInfrastructureReconciled)
 		deployCloudProviderConfig               = f.AddTask(hybridBotanist.DeployCloudProviderConfig, defaultRetry, deployInfrastructure)
 		deployKubeAPIServer                     = f.AddTask(hybridBotanist.DeployKubeAPIServer, defaultRetry, deploySecrets, deployETCD, waitUntilKubeAPIServerServiceIsReady, deployCloudProviderConfig)
+		_                                       = f.AddTaskConditional(botanist.DeployKubeAPIServerHPA, defaultRetry, isBigCluster)
+		_                                       = f.AddTaskConditional(botanist.DeleteKubeAPIServerHPA, defaultRetry, !isBigCluster)
```

Comment:
No extra function call in the flow is needed, please handle that within the `DeployKubeAPIServer` function itself (as discussed above: move the HPA manifest into the kube-apiserver Helm chart and control via values whether it should get deployed or not).

Handle there as well whether the HPA object must be deleted.